QuantLib/ql/termstructures/yield/discountcurve.hpp

```diff
@@ -113,7 +114,12 @@ namespace QuantLib {
 
     template <class T>
     inline Date InterpolatedDiscountCurve<T>::maxDate() const {
-        return dates_.back();
+        // +7 is just a patch to partially (but safely) deal with a pillar date
+        // possibly not being equal to BootstrapHelper's maturity or latest
+        // relevant date. A better (non-backward-compatible) solution would be
+        //if (maxDate_ != Date())
+        //    return maxDate_;
```

Comment:
@fametrano , @paolomazzocchi : I've made the `maxDate_` change above. Please check out my master and verify that everything still works.